lib/inv_multi_cg_quda.cpp

```diff
@@ -122,183 +119,152 @@ namespace quda {
 
   MultiShiftCG::MultiShiftCG(const DiracMatrix &mat, const DiracMatrix &matSloppy, SolverParam &param,
                              TimeProfile &profile) :
-    MultiShiftSolver(mat, matSloppy, param, profile)
+    MultiShiftSolver(mat, matSloppy, param, profile),
+    init(false)
   {
   }
 
-  MultiShiftCG::~MultiShiftCG() { }
+  void MultiShiftCG::create(std::vector<ColorSpinorField> &x, const ColorSpinorField &b,
+                            std::vector<ColorSpinorField> &p)
+  {
+    if (!init) {
+      profile.TPSTART(QUDA_PROFILE_INIT);
+      MultiShiftSolver::create(x, b);
+      num_offset = param.num_offset;
+
+      reliable = false;
+      for (int j = 0; j < num_offset; j++)
+        if (param.tol_offset[j] < param.delta) reliable = true;
+
+      r = b;
+
+      ColorSpinorParam csParam(b);
+      csParam.create = QUDA_NULL_FIELD_CREATE;
+
+      mixed = param.precision_sloppy != param.precision;
+      group_update = false; // do we want to do group updates regardless of precision?
+
+      x_sloppy.resize(num_offset);
+      if ((mixed && param.use_sloppy_partial_accumulator) || group_update) {
```

Comment:
Is this a new feature, or just a simpler way to represent some logic in the code that's been removed?

Reply:
Mostly a cleanup, and a cleaner way to represent.